tests/datatypes/test_set.py

```diff
@@ -162,37 +185,99 @@ def test_remove(self):
             """)
 
     def test_union(self):
+        # success
         self.assertCodeExecution("""
             x = {1, 2, 3}
             y = {3, 4, 5}
+            w = frozenset([6,8,23,2,4])
             z = x.union(y)
-            print(x)
-            print(y)
-            print(z)
+            t = x.union([6,8,23],[2,7,5],w,x)
+            l = x.union(w)
+            print(sorted(x))
+            print(sorted(y))
+            print(sorted(z))
+            print(sorted(w))
+            print(sorted(t))
+            print(sorted(l))
+            """)
+
+        # failure
+        self.assertCodeExecution("""
+            x = {1, 2, 3}
+            try:
+                z = x.union(7)
+            except TypeError as err:
+                print(err)
+            try:
+                t = x.union([6,8,23],[2,7,5],9,x)
+            except TypeError as err:
+                print(err)
             """)
 
     def test_difference_update(self):
+        # success
         self.assertCodeExecution("""
             x = {1, 2, 3}
             y = {3, 4, 5}
-            x.difference_update(y)
-            print(x)
+            x.difference_update(y,"abc",[7,8,9],(3,7,6))
+            print(sorted(x))
+            """)
+
+        # not iterable test
+        self.assertCodeExecution("""
+            x = set([1, 2, 3])
+            try:
+                x.difference_update(1)
+            except TypeError as err:
+                print(err)
+            try:
+                x.difference_update([3,4,5], 1 , (4,6,2))
+            except TypeError as err:
+                print(err)
             """)
 
     def test_intersection_update(self):
+        # successful
         self.assertCodeExecution("""
             x = {1, 2, 3}
             y = {3, 4, 5}
-            x.intersection_update(y)
-            print(x)
+            x.intersection_update(y,[1,5,7],(4,7,0))
+            print(sorted(x))
+            """)
+
+        # not iterable test
+        self.assertCodeExecution("""
+            x = set([1, 2, 3])
+            try:
+                x.intersection_update(1)
+            except TypeError as err:
+                print(err)
+            try:
+                x.intersection_update([3,4,5], 1 , (4,6,2))
+            except TypeError as err:
+                print(err)
             """)
 
     def test_update(self):
+        # successful
         self.assertCodeExecution("""
             x = {1, 2, 3}
             y = {3, 4, 5}
-            print(x.update(y))
-            print(x)
```

Comment:
No - you've made *a* change, but the original test has still been removed. Again - *any* change that *deletes* tests needs to make a *very* strong argument for that removal. The whole purpose of a test suite is to add tests that exercise functionality that you've explicitly implemented, so that any bugs won't re-occur. If you delete tests, you're removing that safety. This isn't just for this particular line - it's for *any* removals in the entire test suite.